src/main/java/org/swisspush/redisques/metrics/MetricsCollector.java

```diff
@@ -45,16 +45,16 @@ public class MetricsCollector {
     private final AtomicLong maxQueueSize = new AtomicLong(0);
     private final AtomicLong queueStateReadyCount = new AtomicLong(0);
     private final AtomicLong queueStateConsumingCount = new AtomicLong(0);
-    private final RedisQues redisQues;
+    private final Map<String, RedisQues.QueueProcessingState> myQueues;
```

Comment:
This still results in a circular dependency. `QueueProcessingState` cannot be part of `RedisQues`